modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue

```diff
@@ -267,10 +285,26 @@ export default {
       this.hoveredWidget = _id;
       this.hoveredNonForeignWidget = nonForeignId;
     },
-    updateWidgetFocused(widgetId) {
-      this.focusedWidget = widgetId;
+    updateWidgetFocused({ _id, focus }) {
+      this.focusedWidget = _id;
       // Attached to window so that modals can see the area is active
-      window.apos.focusedWidget = widgetId;
+      window.apos.focusedWidget = _id;
+      if (focus) {
+        this.$nextTick(() => {
+          const $el = document.querySelector(`[data-apos-widget-id="${_id}"]`);
+          const headerOffset = window.apos.adminBar.height;
+          const elementPosition = $el.getBoundingClientRect().top;
+          const offsetPosition = elementPosition + window.pageYOffset - (headerOffset + 40);
+
+          $el.focus({
+            preventScroll: true
+          });
+          window.scrollTo({
+            top: offsetPosition,
+            behavior: 'smooth'
+          });
+        });
+      }
```

Comment:
```suggestion
      if (scrollIntoView) {
        this.$nextTick(() => {
          const $el = document.querySelector(`[data-apos-widget-id="${_id}"]`);
          if (!$el) {
            return;
          }

          $el.focus({
            preventScroll: true
          });
          $el.scrollIntoView({
            behavior: 'smooth'
          });
        });
      }
```
Not sure all the extra offset position is really worth it.

Reply:
respectfully keeping this